preppal-fe/src/pages/recipe-calendar.tsx

```diff
@@ -0,0 +1,236 @@
+  const [username, setName] = React.useState("");
+  const [ownRecipes, setOwnRecipes] = React.useState<any[]>([]);
+  const [calendarObject, setCalendarObject] = React.useState<any>();
+  const [recipeID, setRecipeID] = React.useState("");
+  const [recipeTitle, setRecipeTitle] = React.useState("");
+  const [shoppingList, setShoppingList] = React.useState("");
+
+  React.useEffect(() => {
+    getUser();
+    getOwnRecipes().then(result => setOwnRecipes(result));
+    // eslint-disable-next-line
+  }, []);
+
+  const getUser = async () => {
+    const token = sessionStorage.getItem("token");
+    try {
+      if (token) {
+        // Get username
+        const req = {
+          method: "GET",
+          headers: {
+            "x-auth-token": token
+          }
+        };
+        const res = await fetch("http://localhost:9001/api/auth/", req).then(res => res.json());
+        setName(res.username);
+
+        // Get Calendar for user
+        const reqCal = {
+          method: "POST",
+          headers: {
+            'Content-Type': 'application/json'
+          },
+
+          body: JSON.stringify({
+            'username': res.username
+          })
+        };
+        const resCal = await fetch("http://localhost:9001/api/calendar/getCalendar", reqCal).then(res => res.json());
+        setCalendarObject(resCal);
+        // set global variables depending on what calendar was returned
+        if (resCal && resCal.calendarDates && resCal.calendarDates.length > 0) {
+          // if calendar exists and some days exist in calendarDates, check if calendarDates exist for today
+          const index = resCal.calendarDates.findIndex((calDate: any) => calDate.dateIs === currDate?.toLocaleString().split(",")[0]);
+          if (index > -1) {
+            // if calendarDates has an object for today, load into globals
+            setRecipeID(resCal.calendarDates[index].recipeOfTheDayID);
+            setRecipeTitle(resCal.calendarDates[index].recipeOfTheDayTitle);
+            setShoppingList(resCal.calendarDates[index].recipeOfTheDayIngredients);
+          }
+          else {
+            // if calendarDates does not have an object for today, default to empty
+            setRecipeID("");
+            setRecipeTitle("");
+            setShoppingList("");
+          }
+        }
+        else {
+          // default to empty
+          setRecipeID("");
+          setRecipeTitle("");
+          setShoppingList("");
+        }
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async function getOwnRecipes() {
+    const token = sessionStorage.getItem("token");
+    let fetchedRecipes: any[] = [];
+    try {
+      if (token) {
+        const req = {
+          method: "GET",
+          headers: {
+            'x-auth-token': token
+          }
+        };
+        fetchedRecipes = await fetch("http://localhost:9001/api/users/ownRecipes/", req).then((res) => res.json());
```

Comment:
Why are we calling `api/users/ownRecipes`?
Would it be better if we just call `api/recipes/` to return a list of all public recipes?

Reply:
Tested using `api/recipes/` and the functionality is the same.